nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/JoltTransformJSON.java

```diff
@@ -332,12 +360,15 @@ public void setup(final ProcessContext context) {
         } catch (final Exception ex) {
             getLogger().error("Unable to setup processor", ex);
         }
+
+        inputCharset = Charset.forName(context.getProperty(INPUT_CHARSET).getValue());
+        outputCharset = Charset.forName(context.getProperty(OUTPUT_CHARSET).getValue());
     }
 
     private JoltTransform createTransform(final ProcessContext context, final String specString) throws Exception {
         final Object specJson;
         if (context.getProperty(JOLT_SPEC).isSet() && !SORTR.getValue().equals(context.getProperty(JOLT_TRANSFORM).getValue())) {
-            specJson = JsonUtils.jsonToObject(specString, DEFAULT_CHARSET);
+            specJson = JsonUtils.jsonToObject(specString, inputCharset.name());
```

Comment:
There is a similar call in the advanced UI that may also need to be addressed for consistency.

Reply:
Hmm I have no idea how to fix that. I have no experience in fixing custom UI pieces in NiFi as I've actively avoided them since I started working with the framework.

Will the Advanced UI respect the processor setting?

Reply:
@travisneeley the advanced UI uses some shared utilities with the processor but not everything, so it would need to be updated as well.  In this case you would need to add the input/output charset entry and ensure that the test processor applies those values to test spec as well as test input/output.

I'd be happy to guide you through edits required and how to test if needed (can provide more guidance in comments on the JIRA).  The important idea is to ensure it's functionality remains consistent with processor functionality.